plugins/ltac/tacinterp.ml

```diff
@@ -1957,7 +1957,8 @@ let lifts f = (); fun ist x -> Ftactic.enter begin fun gl ->
   let sigma = Proofview.Goal.sigma gl in
   let (sigma, v) = f ist env sigma x in
   Proofview.tclTHEN (Proofview.Unsafe.tclEVARS sigma)
-  (Ftactic.return v)
+    (Proofview.tclTHEN (Proofview.Unsafe.tclSETENV (Global.env()))
+       (Ftactic.return v))
```

Comment:
Please add a FIXME comment here.

Reply:
Is there a need for a fixme? It seems like every tclSETENV will go away if we fix side effects, and won't if we don't.

Reply:
`tclSETENV` is legit, the dreaded call is `tclSETENV + Global.env ()`.

Reply:
OK, comment added.